data-prepper-plugins/saas-source-plugins/source-crawler/src/main/java/org/opensearch/dataprepper/plugins/source/source_crawler/coordination/scheduler/WorkerScheduler.java

```diff
@@ -75,13 +96,31 @@ public void run() {
         log.warn("SourceItemWorker Scheduler is interrupted, looks like shutdown has triggered");
     }
 
-    private void processPartition(EnhancedSourcePartition partition, Buffer<Record<Event>> buffer, CrawlerSourceConfig sourceConfig) {
+    private void processPartition(EnhancedSourcePartition partition, Buffer<Record<Event>> buffer) {
         // Implement your source extraction logic here
         // Update the partition state or commit the partition as needed
         // Commit the partition to mark it as processed
         if (partition.getProgressState().isPresent()) {
-            crawler.executePartition((SaasWorkerProgressState) partition.getProgressState().get(), buffer, sourceConfig);
+            AcknowledgementSet acknowledgementSet = null;
+            if (sourceConfig.isAcknowledgments()) {
+                acknowledgementSet = createAcknowledgementSet(partition);
+            }
+            crawler.executePartition((SaasWorkerProgressState) partition.getProgressState().get(), buffer, acknowledgementSet);
         }
         sourceCoordinator.completePartition(partition);
     }
+
+    private AcknowledgementSet createAcknowledgementSet(EnhancedSourcePartition partition) {
+        return acknowledgementSetManager.create((result) -> {
+            if (result) {
+                acknowledgementSetSuccesses.increment();
+                sourceCoordinator.completePartition(partition);
+                log.debug("acknowledgements received for partitionKey: {}", partition.getPartitionKey());
+            } else {
+                acknowledgementSetFailures.increment();
+                log.debug("acknowledgements received with false for partitionKey: {}", partition.getPartitionKey());
+                sourceCoordinator.giveUpPartition(partition);
+            }
+        }, ACKNOWLEDGEMENT_SET_TIMEOUT);
```

Comment:
Wondering how do you choose 20s as default? Do you think it's necessary to make it user configurable?